### PR TITLE
gh-87304: Summary of the changes made: added detailed explanation for import

### PR DESCRIPTION
**Pull request by robert861212**
Explained how import works in better details in the doc. Mentored by Toshio Kuratomi.

#87304

### CODE REVIEW

File: Doc/reference/simple_stmts.rst [hunk 1]
```diff
@@ -806,6 +806,14 @@ Examples::
    from foo.bar import baz    # foo.bar.baz imported and bound as baz
    from foo import attr       # foo imported and foo.attr bound as attr
 
+Note: When a submodule is imported, all parent modules are imported and may
```
Comment (JelleZijlstra) [inline Doc/reference/simple_stmts.rst]: I think this is too much detail. Above there is already a reference to https://docs.python.org/3/reference/import.html#searching, which explains this behavior. I would just change the comments in the above block to say "foo, foo.bar, and foo.bar.baz imported".

File: Doc/reference/simple_stmts.rst [hunk 2]
```diff
@@ -801,10 +801,10 @@ The :keyword:`from` form uses a slightly more complex process:
...
 
    import foo                 # foo imported and bound locally
-   import foo.bar.baz         # foo.bar.baz imported, foo bound locally
-   import foo.bar.baz as fbb  # foo.bar.baz imported and bound as fbb
-   from foo.bar import baz    # foo.bar.baz imported and bound as baz
-   from foo import attr       # foo imported and foo.attr bound as attr
+   import foo.bar.baz         # foo, foo.bar, and foo.bar.baz imported, foo bound locally
+   import foo.bar.baz as fbb  # foo, foo.bar, and foo.bar.baz imported, foo.bar.baz bound as fbb
+   from foo.bar import baz    # foo, foo.bar, and foo.bar.baz imported, foo.bar.baz bound as baz
+   from foo import attr       # foo and foo.attr imported and foo.attr bound as attr
```
Comment (JelleZijlstra) [inline Doc/reference/simple_stmts.rst]:
```suggestion
   from foo import attr       # foo imported and foo.attr bound as attr
```
I think the intent here is that `foo.attr` is a regular attribute of `foo`, not a submodule.

Comment (robert861212) [reply Doc/reference/simple_stmts.rst]: Would foo.attr still be technically imported?

Comment (JelleZijlstra) [reply Doc/reference/simple_stmts.rst]: No, see line 793 above. If the attr already exists, it doesn't try to import it.

Comment (robert861212) [reply Doc/reference/simple_stmts.rst]: got it! refactored

**Review by JelleZijlstra — APPROVED** [review]

**Comment by JelleZijlstra** [comment]
Thanks for your contribution!

**Pull request merged**